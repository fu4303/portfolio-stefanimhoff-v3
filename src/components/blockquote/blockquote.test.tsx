import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Blockquote } from '.';
import type { BlockquoteProps } from '.';

type BlockquoteTestProps = BlockquoteProps & {
  description: string;
};

describe('Blockquote', () => {
  // prettier-ignore
  const combinations: BlockquoteTestProps[] = [
    { description: 'default', children: 'Quote' },
    { description: 'german', children: 'Quote', lang: 'de' },
    { description: 'japanese', children: 'Quote', lang: 'ja' },
    { description: 'author', children: 'Quote', author: 'Author' },
    { description: 'author, source', children: 'Quote', author: 'Author', source: 'Source' },
    { description: 'author, sourceUrl', children: 'Quote', author: 'Author', sourceUrl: 'SourceUrl' },
    { description: 'source', children: 'Quote', source: 'Source' },
    { description: 'sourceUrl', children: 'Quote', sourceUrl: 'SourceUrl' },
    { description: 'source, sourceUrl', children: 'Quote', source: 'Source', sourceUrl: 'SourceUrl' },
    { description: 'author, source, sourceUrl', children: 'Quote', author: 'Author', source: 'Source', sourceUrl: 'SourceUrl' },
  ];

  combinations.forEach((combination) => {
    test(`renders in ${combination.description}`, () => {
      const { container } = render(<Blockquote {...combination} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  combinations.forEach((combination) => {
    test(`is accessible in ${combination.description}`, async () => {
      const { container } = render(<Blockquote {...combination} />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
