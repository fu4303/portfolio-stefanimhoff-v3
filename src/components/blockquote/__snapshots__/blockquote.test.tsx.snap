// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders in author 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <b
      class="c2"
    >
      Author
    </b>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in author, source 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <b
      class="c2"
    >
      Author
    </b>
    , 
    <cite
      class=""
    >
      Source
    </cite>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in author, source, sourceUrl 1`] = `
.c3 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <b
      class="c2"
    >
      Author
    </b>
    , 
    <cite
      class=""
    >
      <a
        class="c3"
        href="SourceUrl"
        rel="nofollow noopener noreferrer external"
      >
        Source
      </a>
    </cite>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in author, sourceUrl 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <b
      class="c2"
    >
      Author
    </b>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in default 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Blockquote renders in german 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="de"
>
  Quote
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Blockquote renders in japanese 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="ja"
>
  Quote
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Blockquote renders in source 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <cite
      class=""
    >
      Source
    </cite>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in source, sourceUrl 1`] = `
.c2 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  >
    —
    <cite
      class=""
    >
      <a
        class="c2"
        href="SourceUrl"
        rel="nofollow noopener noreferrer external"
      >
        Source
      </a>
    </cite>
  </footer>
</blockquote>
`;

exports[`Blockquote renders in sourceUrl 1`] = `
.c0 {
  font-style: italic;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  margin-bottom: clamp( 0.833rem,1.78vw,1.0rem);
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
}

<blockquote
  class="c0"
  lang="en"
>
  Quote
  <footer
    class="c1"
  />
</blockquote>
`;
